apps/frontend/src/pages/CreateListing.tsx

```diff
@@ -1,5 +1,281 @@
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Container,
+  FormControl,
+  Grid,
+  TextField,
+  Typography,
+} from "@mui/material";
+import { useCallback, useEffect, useState } from "react";
+import _axios_instance from "../_axios_instance.tsx";
+import { colors } from "../styles/colors.tsx";
+import MultiFileUpload from "../extra_components/MultiFileUpload.tsx";
+import Carousel from "../extra_components/Carousel.tsx";
+
+interface URLObject {
+  url: string;
+}
+interface LocationObject {
+  latitude: number;
+  longitude: number;
+}
+
+interface ListingObject {
+  title: string;
+  description: string;
+  price: number;
+  location: LocationObject;
+  images: URLObject[];
+}
+
+interface NewListingObject {
+  listing: ListingObject;
+}
+
 const CreateListing = () => {
-  return <h1>CreateListing</h1>;
+  const [images, setImages] = useState<string[]>([]);
+
+  // This newListingObject bundles all the listing data for upload to the server
+  const [newListingObject, setNewListingObject] = useState<NewListingObject>({
+    listing: {
+      title: "",
+      description: "",
+      price: 0,
+      location: {
+        latitude: 0.0,
+        longitude: 0.0,
+      },
+      images: [],
+    },
+  });
+
+  // Gets the user location, and adds it to the listing object
+  const getUserLocation = useCallback(async () => {
+    try {
+      navigator.geolocation.getCurrentPosition((position) => {
+        newListingObject.listing.location.longitude = position.coords.longitude;
```

Comment:
I think this is modifying the properties of a const variable

Reply:
I tried modifying this with useState before, but despite it working for updating my newListingObject in other places it just doesn't work seeming randomly in other places

```ts
setNewListingObject((prevState) => ({
          ...prevState,
          listing: {
            ...prevState.listing,
            location: {
              ...prevState.listing.location,
              latitude = position.coords.latitude,
              longitude = position.coords.longitude,
            },
          },
        }));
```

Reply:
It's pretty important to use useState at all times when modifying a variable in state, it's probably the only thing in this PR that I would insist is fixed before approving. You could try something along the lines of

`setNewListingObject(
  {...newListingObject,
    listing: {
      ...newListingObject.listing,
      location: {
        <location stuff>
      }
    }
  }
)`

What you have there should work, but if it doesn't trying this might simplify it. If you're updating the object without using useState though, that would likely be the cause of errors when you do try and call useState